backend/infrahub/core/query/relationship.py

```diff
@@ -948,3 +956,147 @@ async def get_count_per_peer(self) -> dict[str, int]:
+                        SET deleted_edge.hierarchy = active_edge.hierarchy
+                        WITH active_edge, n
+                        WHERE active_edge.branch = $branch AND active_edge.to IS NULL
+                        SET active_edge.to = $at
+                    }
+                """ % {
+                    "arrow_left": arrow_left,
+                    "arrow_right": arrow_right,
+                    "active_rel_filter": active_rel_filter,
+                    "edge_type": edge_type,
+                }
+
+                query += sub_query
+
+        # We only want to return uuid/kind of `Node` connected through `IS_RELATED` edges.
+        is_related_subqueries = [
+            """
+                WITH rl
+                MATCH (rl)%(arrow_left)s[active_edge:%(edge_type)s]%(arrow_right)s(n)
+                WHERE %(active_rel_filter)s AND active_edge.status ="active"
+                CREATE (rl)%(arrow_left)s[deleted_edge:%(edge_type)s $rel_prop]%(arrow_right)s(n)
+                SET deleted_edge.hierarchy = active_edge.hierarchy
+                WITH rl, active_edge, n
+                WHERE active_edge.branch = $branch AND active_edge.to IS NULL
+                SET active_edge.to = $at
+                RETURN n.uuid as uuid, n.kind as kind, rl.name as rel_identifier
+        """
+            % {
+                "arrow_left": arrow_left,
+                "arrow_right": arrow_right,
+                "active_rel_filter": active_rel_filter,
+                "edge_type": DatabaseEdgeType.IS_RELATED.value,
+            }
+            for arrow_left, arrow_right in (("<-", "-"), ("-", "->"))
+        ]
+
+        sub_query = """
+        UNION
+        """.join(is_related_subqueries)
+
+        sub_query = (
+            """
+        CALL {
+        """
+            + sub_query
+            + """
+        }
+        RETURN DISTINCT uuid, kind, rel_identifier
+        """
+        )
+
+        query += sub_query
+
+        self.add_to_query(query)
+
+    def get_deleted_relationships_changelog(
+        self, node_schema: NodeSchema
+    ) -> list[RelationshipCardinalityOneChangelog | RelationshipCardinalityManyChangelog]:
+        rel_identifier_to_changelog_mapper = {}
+
+        for result in self.get_results():
+            peer_uuid = result.data["uuid"]
+            if peer_uuid == self.node_id:
+                continue
+
+            rel_identifier = result.data["rel_identifier"]
+            kind = result.data["kind"]
+            deleted_rel_schemas = [
+                rel_schema for rel_schema in node_schema.relationships if rel_schema.identifier == rel_identifier
+            ]
+            if len(deleted_rel_schemas) > 1:
+                raise ValueError(f"Duplicated relationship schema with identifier {rel_identifier}")
+            if len(deleted_rel_schemas) == 0:
+                continue  # TODO Unidirectional relationship changelog should be handled.
```

Comment:
I added some logic here to handle relationships changelog regarding how we delete relationships now (using a single query instead of RelationshipManager).

I will create an issue to handle missing unidirectional relationships changelogs (which is currently not supported anyway), but I'd rather merge this PR asap to avoid blocking merging stable changes into release-1.2.